packages/amplify-category-api/src/index.ts

```diff
@@ -180,55 +198,72 @@ export async function getPermissionPolicies(context, resourceOpsMapping) {
           permissionPolicies.push(policy);
           resourceAttributes.push({ resourceName, attributes, category });
         } else {
-          context.print.error(`Provider not configured for ${category}: ${resourceName}`);
+          printer.error(`Provider not configured for ${category}: ${resourceName}`);
         }
       } catch (e) {
-        context.print.warning(`Could not get policies for ${category}: ${resourceName}`);
+        printer.warn(`Could not get policies for ${category}: ${resourceName}`);
         throw e;
       }
     }),
   );
   return { permissionPolicies, resourceAttributes };
 }
 
-export async function executeAmplifyCommand(context) {
+export async function executeAmplifyCommand(context: $TSContext) {
   let commandPath = path.normalize(path.join(__dirname, 'commands'));
   if (context.input.command === 'help') {
     commandPath = path.join(commandPath, category);
   } else {
     commandPath = path.join(commandPath, category, context.input.command);
   }
 
-  const commandModule = require(commandPath);
-  await commandModule.run(context);
+  const commandModule = await import(commandPath);
+  try {
+    await commandModule.run(context);
+  } catch (error) {
```

Comment:
aren't errors caught and handled higher up the call stack? Doesn't seem like we need a try/catch here

Reply:
Prior to this, these errors were being handled in each individual command. This retains mostly the same behavior, while standardizing the precise error handling.